### PR TITLE
Remove NSMutableArray for retaining sublayout elements

### PR DESCRIPTION
**Pull request by Adlai-Holler**
A little optimization.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/b1ad97dd3494a307f54b07899f7e524b/raw/eda965a73a2c76bba0570b50cbea6d21946b87b0/log.txt)

File: Source/Layout/ASLayout.mm [hunk 1]
```diff
@@ -134,18 +129,14 @@ - (instancetype)initWithLayoutElement:(id<ASLayoutElement>)layoutElement
...
   
   return self;
 }
 
-- (instancetype)init
-{
-  ASDisplayNodeAssert(NO, @"Use the designated initializer");
-  return [self init];
-}
-
```
Comment (Adlai-Holler) [inline Source/Layout/ASLayout.mm]: Needed to kick the CI so I tossed this in.

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/c29a46c7c2c4020b980efc89aa6e13db/raw/822eff0cab1a782e23b6e1b608a8cce34cc6f807/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/db8fff04528f46d00a69521c7f66dc07/raw/5beba724a698940fdcde385cf8f2ba758da06ac2/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/4931a04a747cb8e5da6bce387c4dc53e/raw/b0b856a738f6f65d3db5c70c99a6486bac8da5a3/log.txt)

**Comment by Adlai-Holler** [comment]
I give up on the CI.

**Pull request merged**